fuel-core/src/service/modules.rs

```diff
@@ -96,12 +99,29 @@ pub async fn start_modules(config: &Config, database: &Database) -> Result<Modul
     }
     txpool.start().await?;
 
+    let p2p_db: Arc<Box<dyn P2pDb>> = Arc::new(Box::new(database.clone()));
```

Comment:
Why do we need the extra box inside the arc (`Arc<Box<dyn T>>`), shouldn't `Arc<dyn T>` suffice? Both Box and Arc allocate on the heap.

Reply:
you're right, guess my mind was already on vacation